### PR TITLE
Optional bias offset subtraction and general optional automatic calibration

### LINKED ISSUE #171: Add NoiseMaps to prescan_biassub for L1 to L2 recipes
**Describe the bug**
I think we should be using NoiseMaps in prescan_biassub for L1 to L2 (a and b) science frame recipes to improve the biass offset subtraction. 

Requires additional logic in walker to handle optional calibration files

**Expected behavior**
Adding them to the recipe and checking we still match the TVAC L2 reductions

**corgidrp version:**
v0.2

### PR DESCRIPTION
**Pull request by semaphoreP**
## Describe your changes

 * Added ability to specify optional automatic calibrations, where `None` can be passed into the step function if no suitable calibration file is found, and the step function can still run
 * Added bias offset subtraction into L1_to_L2b e2e recipe, and optionally into L1_to_l2a recipe. Now requires NoiseMaps for L1 to L2a reduction if you want this. I have updated the pipeline implementation doc table to specify this as such. 
 * Checked the e2e tests still run

## Type of change

Please delete options that are not relevant (and this line).

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## Reference any relevant issues (don't forget the #)

Closes #171

## Checklist before requesting a review
- [x] I have linted my code
- [x] I have verified that all unit tests pass in a clean environment and added new unit tests, as needed
- [x] I have verified that all docstrings are properly formatted and added new documentation, as needed

### CODE REVIEW

File: tests/test_walker.py
Comment (maxwellmb) [inline tests/test_walker.py:1]: I don't think we're testing for the case where `skip_missing_cal_steps` is `False` and the optional calib isn't there, are we?

File: corgidrp/recipe_templates/l1_to_l2b.json [hunk 1]
```diff
@@ -9,6 +9,9 @@
     "steps" : [
         {
             "name" : "prescan_biassub",
+            "calibs" : {
+                "DetectorNoiseMaps" : "AUTOMATIC"
```
Comment (maxwellmb) [inline corgidrp/recipe_templates/l1_to_l2b.json:13]: why isn't this one OPTIONAL, but it is in `l1_to_l2a_eng.json` and `l1_to_l2a_basic.json`?

**Review by maxwellmb — CHANGES_REQUESTED** [review]
There are two main comments here: 

- Is there a missing OPTIONAL in one of the recipes
- Are the tests complete?

File: tests/test_walker.py
Comment (semaphoreP) [reply tests/test_walker.py:1]: Just added this test

**Comment by semaphoreP** [comment]
@maxwellmb ready for re-review

**Review by maxwellmb — APPROVED** [review]

**Pull request merged**